cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/GetRevisionTaskThreaded.java

```diff
@@ -137,14 +142,36 @@ public boolean add(Future<DocumentRevsList> documentRevsListFuture) {
         // we make the request at construction time...
         for (final BulkGetRequest request : requests) {
 
-            completionService.submit(new Callable<DocumentRevsList>() {
+            submittedJobs.add(completionService.submit(new Callable<DocumentRevsList>() {
                 @Override
                 public DocumentRevsList call() throws Exception {
                     return new DocumentRevsList(GetRevisionTaskThreaded.this.sourceDb.getRevisions
                             (request.id, request.revs, request.atts_since,
                                     GetRevisionTaskThreaded.this.pullAttachmentsInline));
                 }
-            });
+            }));
+        }
+    }
+
+    void cancel() {
+        synchronized (submittedJobs) {
```

Comment:
You don't need this `synchronized` block because `submittedJobs` was already constructed with `synchronizedSet`